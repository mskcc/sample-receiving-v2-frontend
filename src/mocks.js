
export const initialFormStateMock = {
  allContainers: [],
  allApplications: [],
  allMaterials: [],
  allPatientIdFormats: [],
  applications: [],
  containers: [],
  formIsLoading: true,
  filteredContainers: [
    { id: 'Plates', value: 'Plates' },
    { id: 'Micronic Barcoded Tubes', value: 'Micronic Barcoded Tubes' },
  ],
  filteredContainersBS: [
    { id: 'Blocks/Slides/Tubes', value: 'Blocks/Slides/Tubes' },
  ],
  materials: [],
  picklists: {},
  selectedApplication: '',
  selectedMaterial: '',
  species: [],
  patientIDTypeNeedsFormatting: false,
}

export const initialGridStateMock = {
  columns: [],
  rows: [],
  error: '',
  form: [],
  gridIsLoading: false,
  nothingToChange: false,
}

export const filledGridStateMock = {
  upload: {
    grid: {
      columns: ['Block/Slide/TubeID', 'Sample ID'],
      rows: [
        { tubeId: '', userId: '' },
        { tubeId: '', userId: '' },
        { tubeId: '', userId: '' },
        { tubeId: '', userId: '' },
      ],
      error: '',
      form: {
        material: 'Tissue',
        application: 'CustomCapture',
        igo_request_id: '444444',
        number_of_samples: '4',
        species: 'Tuberculosis',
        container: 'Plates',
        patient_id_type: '',
      },
      gridIsLoading: false,
      nothingToChange: false,
      columnFeatures: [
        {
          columnHeader: 'Block/Slide/TubeID',
          data: 'tubeId',
        },
        {
          columnHeader: 'Sample ID',
          data: 'userId',
        },
      ],
    },
  },
}

export const initialFullStateMock = {
  upload: {
    form: {
      allContainers: [],
      allApplications: [],
      allMaterials: [],
      allPatientIdFormats: [],
      applications: [],
      containers: [],
      formIsLoading: false,
      filteredContainers: [
        { id: 'Plates', value: 'Plates' },
        { id: 'Micronic Barcoded Tubes', value: 'Micronic Barcoded Tubes' },
      ],
      filteredContainersBS: [
        { id: 'Blocks/Slides/Tubes', value: 'Blocks/Slides/Tubes' },
      ],
      materials: [],
      selectedApplication: '',
      selectedMaterial: '',
      species: [],
      patientIDTypeNeedsFormatting: false,
    },
    grid: {
      columns: [],
      rows: [],
      error: '',
      form: [],
      gridIsLoading: false,
      nothingToChange: false,
    },
  },
}

export const filledFullStateMock = {
  upload: {
    form: {
      allContainers: [{ id: 'test', value: 'test' }],
      allApplications: [{ id: 'test', value: 'test' }],
      allMaterials: [{ id: 'test', value: 'test' }],
      allPatientIdFormats: [{ id: 'test', value: 'test' }],
      applications: [{ id: 'test', value: 'test' }],
      containers: [{ id: 'test', value: 'test' }],
      formIsLoading: false,
      filteredContainers: [
        { id: 'Plates', value: 'Plates' },
        { id: 'Micronic Barcoded Tubes', value: 'Micronic Barcoded Tubes' },
      ],
      filteredContainersBS: [
        { id: 'Blocks/Slides/Tubes', value: 'Blocks/Slides/Tubes' },
      ],
      materials: [{ id: 'test', value: 'test' }],
      selectedApplication: '',
      selectedMaterial: '',
      species: [{ id: 'test', value: 'test' }],
      patientIDTypeNeedsFormatting: false,
    },
    grid: {
      columns: ['Block/Slide/TubeID', 'Sample ID'],
      rows: [{ tubeId: '', userId: '' }, { tubeId: '', userId: '' }],
      error: '',
      form: {
        material: 'Tissue',
        application: 'CustomCapture',
        igo_request_id: '444444',
        number_of_samples: '4',
        species: 'Tuberculosis',
        container: 'Plates',
        patient_id_type: '',
      },
      gridIsLoading: false,
      nothingToChange: false,
      columnFeatures: [
        {
          columnHeader: 'Block/Slide/TubeID',
          data: 'tubeId',

          name: 'Block/Slide/TubeID',
        },
        {
          columnHeader: 'Sample ID',
          data: 'userId',
        },
      ],
    },
  },
}

export const formValuesMock = {
  application: 'CustomCapture',
  container: 'Plates',
  igo_request_id: '444444',
  material: 'Tissue',
  number_of_samples: '2',
  patient_id_type: '',
  species: 'Tuberculosis',
}

export const columnDefsResponseMock = {
  columnDefs: [
    {
      columnHeader: 'Block/Slide/TubeID',
      data: 'tubeId',
      editableCellTemplate:
        '<div><form name="inputForm"><input class="form-control inputheight" type="INPUT_TYPE" ng-class="\'colt\' + col.uid" ui-grid-editor ng-model="MODEL_COL_FIELD" ng-paste="grid.appScope.handleCellPaste($event)"></form></div>',
      headerCellClass: 'required',
      helpText:
        'The identifier on your tube, block or slide.  You can paste in directly from excel, and there are no formatting rules.  Please be as correct as possible, and ensure your tubes, blocks and slides are labeled clearly.',
      name: 'Block/Slide/TubeID',
    },
    {
      columnHeader: 'Sample ID',
      data: 'userId',
      editableCellTemplate:
        '<div><form name="inputForm"><input class="form-control inputheight" type="INPUT_TYPE" ng-class="\'colt\' + col.uid" ui-grid-editor ng-model="MODEL_COL_FIELD" ng-paste="grid.appScope.handleCellPaste($event)"></form></div>',
      headerCellClass: 'required',
      helpText:
        'The Sample ID stays with your sample for its lifetime. Letters, numbers, dashes, and underscores only, three char min. You cannot have more than one underscore consecutively.',
      name: 'Sample ID',
    },
    {
      columnHeader: 'Species',
      data: 'organism',
      editableCellTemplate: 'ui-grid/dropdownEditor',
      headerCellClass: 'required',
      helpText:
        'If your species is not available, please contact IGO immediately',
      name: 'Species',
      picklistName: 'Species',
      source: [
        { id: 'Human', value: 'Human' },
        { id: 'Mouse', value: 'Mouse' },
        { id: 'Mouse_GeneticallyModified', value: 'Mouse_GeneticallyModified' },
        { id: 'Drosophilia', value: 'Drosophilia' },
        { id: 'Zebrafish', value: 'Zebrafish' },
        { id: 'Chicken', value: 'Chicken' },
        { id: 'Bacteria', value: 'Bacteria' },
        { id: 'S.Cerevisae', value: 'S.Cerevisae' },
        { id: 'Tuberculosis', value: 'Tuberculosis' },
        { id: 'E.Coli ', value: 'E.Coli ' },
        { id: 'C.Elegans', value: 'C.Elegans' },
        { id: 'S.Pombe', value: 'S.Pombe' },
        { id: 'R.norvegicus', value: 'R.norvegicus' },
        { id: 'R.rattus', value: 'R.rattus' },
        { id: 'Plasmid', value: 'Plasmid' },
        { id: 'Archaea', value: 'Archaea' },
        { id: 'other', value: 'other' },
      ],
      type: 'dropdown',
      trimDropdown: false,
    },
  ],
}

export const gridMock = {
  columnFeatures: [
    {
      columnHeader: 'Block/Slide/TubeID',
      data: 'tubeId',
      editableCellTemplate: `<div><form name="inputForm"><input class="form-control inputheight" type="INPUT_TYPE" ng-class="'colt' + col.uid" ui-grid-editor ng-model="MODEL_COL_FIELD" ng-paste="grid.appScope.handleCellPaste($event)"></form></div>`,
      headerCellClass: 'required',
      helpText:
        'The identifier on your tube, block or slide.  You can paste in directly from excel, and there are no formatting rules.  Please be as correct as possible, and ensure your tubes, blocks and slides are labeled clearly.',
      name: 'Block/Slide/TubeID',
    },
    {
      columnHeader: 'Sample ID',
      data: 'userId',
      editableCellTemplate: `<div><form name="inputForm"><input class="form-control inputheight" type="INPUT_TYPE" ng-class="'colt' + col.uid" ui-grid-editor ng-model="MODEL_COL_FIELD" ng-paste="grid.appScope.handleCellPaste($event)"></form></div>`,
      headerCellClass: 'required',
      helpText:
        'The Sample ID stays with your sample for its lifetime. Letters, numbers, dashes, and underscores only, three char min. You cannot have more than one underscore consecutively.',
      name: 'Sample ID',
    },
    {
      columnHeader: 'Species',
      data: 'organism',
      editableCellTemplate: 'ui-grid/dropdownEditor',
      headerCellClass: 'required',
      helpText:
        'If your species is not available, please contact IGO immediately',
      name: 'Species',
      picklistName: 'Species',
      source: [
        'Human',
        'Mouse',
        'Mouse_GeneticallyModified',
        'Drosophilia',
        'Zebrafish',
        'Chicken',
        'Bacteria',
        'S.Cerevisae',
        'Tuberculosis',
        'E.Coli ',
        'C.Elegans',
        'S.Pombe',
        'R.norvegicus',
        'R.rattus',
        'Plasmid',
        'Archaea',
        'other',
      ],
      type: 'dropdown',
      trimDropdown: false,
    },
  ],
  columnHeaders: ['Block/Slide/TubeID', 'Sample ID', 'Species'],
  rows: [
    {
      tubeId: '',
      userId: '',
      organism: 'Tuberculosis',
    },
    {
      tubeId: '',
      userId: '',
      organism: 'Tuberculosis',
    },
  ],
}
