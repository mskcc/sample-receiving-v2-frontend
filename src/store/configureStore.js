import { createStore, applyMiddleware, compose } from 'redux'
import { Provider } from 'react-redux'

import thunk from 'redux-thunk'
import multi from 'redux-multi'
import { persistStore } from 'redux-persist'
import { createLogger } from 'redux-logger'
import rootReducer from '../reducers/rootReducer'
import DevTools from '../containers/DevTools'

const middleware = [thunk, multi]
if (process.env.NODE_ENV === 'production') {
}

const store = createStore(
  rootReducer,
  compose(
    applyMiddleware(...middleware),
    DevTools.instrument()
  )
)


let persistor = persistStore(store)

export {store, persistor}
