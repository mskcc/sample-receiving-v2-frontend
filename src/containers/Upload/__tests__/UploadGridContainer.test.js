import React, { Component } from 'react'

import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import multi from 'redux-multi'

import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import TestUploadGridContainer from '../UploadGridContainer'

import { Provider } from 'react-redux'

import { initialFullStateMock, gridMock } from '../../../mocks'

// const formTestStore = initialFormStateMock

const middlewares = [thunk, multi]
const mockStore = configureStore(middlewares)

const gridTestStore = mockStore({
  ...initialFullStateMock,
  upload: {
    ...initialFullStateMock.upload,
    grid: {
      ...initialFullStateMock.upload.grid,
      ...gridMock,
      columns: gridMock.columnHeaders,
    },
  },
})

const UploadGridContainer = props => (
  <Provider store={props.store}>
    <TestUploadGridContainer />
  </Provider>
)

describe('Render UploadGridContainer', () => {
  it('renders UploadGridContainer correctly', () => {
    const UploadGridContainerComponent = renderer
      .create(<UploadGridContainer store={mockStore(initialFullStateMock)} />)
      .toJSON()
    expect(UploadGridContainerComponent).toMatchSnapshot()
  })
})

describe('renders UploadGridContainer', () => {
  it('renders when rows are not empty', () => {
    let container = document.createElement('DIV')
    container.id = 'hotContainer'
    document.body.appendChild(container)
    const wrapper = mount(<UploadGridContainer store={gridTestStore} />, {
      attachTo: document.getElementById('hotContainer'),
    })

    expect(wrapper.find('HotTable').exists()).toBe(true)
  })
})
