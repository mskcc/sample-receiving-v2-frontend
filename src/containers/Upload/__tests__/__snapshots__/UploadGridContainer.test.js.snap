// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render UploadGridContainer renders UploadGridContainer correctly 1`] = `null`;
